tests/pkg_fatfs_vfs/main.c

```diff
@@ -292,9 +292,83 @@ static void test_create(void)
     nw = vfs_write(fd, test_txt, sizeof(test_txt));
     print_test_result("test_create__write_wo", nw == sizeof(test_txt));
     print_test_result("test_create__close_wo", vfs_close(fd) == 0);
+
+    /* test create if file exists */
+    fd = vfs_open(FULL_FNAME1, O_WRONLY | O_CREAT, 0);
+    print_test_result("test_create__open_wo2", fd >= 0);
+
+    nw = vfs_write(fd, test_txt, sizeof(test_txt));
+    print_test_result("test_create__write_wo2", nw == sizeof(test_txt));
+    print_test_result("test_create__close_wo2", vfs_close(fd) == 0);
+
     print_test_result("test_create__umount", vfs_umount(&_test_vfs_mount) == 0);
 }
 
+#ifdef MODULE_NEWLIB
+static void test_newlib(void)
+{
+    FILE* fl;
+    char buf[sizeof(test_txt) + sizeof(test_txt2)];
+    print_test_result("test_newlib__mount", vfs_mount(&_test_vfs_mount) == 0);
+
+    /* try to open file that doesn't exist */
+    fl = fopen(FULL_FNAME_NXIST, "r");
+    print_test_result("test_newlib__fopen", fl == NULL);
+    if (fl) {
+        fclose(fl);
+    }
+
+    /* create new file write and check content */
+    remove(FULL_FNAME2);
+    fl = fopen(FULL_FNAME2, "w+");
+    print_test_result("test_newlib__fopen_w", fl != NULL);
+    if (fl) {
+        print_test_result("test_newlib__fputs_w", fputs(test_txt, fl) >= 0);
+        rewind(fl);
+        print_test_result("test_newlib__fread_w",
+                fread(buf, sizeof(*buf), sizeof(buf), fl) > 0);
+        print_test_result("test_newlib__strcmp_w", strcmp(test_txt, buf) == 0);
+        print_test_result("test_newlib__fclose_w", fclose(fl) == 0);
+    }
+
+    /* open file RO */
+    fl = fopen(FULL_FNAME2, "r");
```

Comment:
> > So let's assume a Cppcheck bug!
> 
> Can we update cppcheck on CI?

Would be easier to suppress this warning for now. @sven-hm please add [`cppcheck``suppression comment](https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md#continuous-integration):

```suggestion
    /* cppcheck-suppress resourceLeak
     * (reason: cppcheck <2.0 reports a false positive here */
    fl = fopen(FULL_FNAME2, "r");   /* open file RO */
```